### DISCUSSION TITLE
Save all user times in DB not just PR

### DISCUSSION THREAD

**Discussion by KristianP26** [CS2Surf/Timer#14] [Ideas]
Save all user times in DB not just PR

**Comment by tslashd** [comment]
What would this be used for? 

Also would this be only the "time" (map run) or the checkpoints associated with that run as well?

**Reply by KristianP26** [reply]
I have had a Lot of issues building SurfTimer Web stats.
It will be good for statistics purposes.
You can let now the player how many times he did the map.
There is lot of things what you can do with such of data.

**Reply by KristianP26** [reply]
I can take a detailed look on DB structure and suggest even more things.

**Reply by tslashd** [reply]
What concerns me about this is the amount of data we would have in the DB if we start storing the previous runs of the players, even if we skip past the point of your PB having a unique `maptime_id` which we currently just update if you set a new PB, each checkpoint data for the run is attached to the said ID (in a different table), a new system would have to be made for storing older runs and "updating" current ones, also bare in mind replays for each run are saved in the same `MapTime` entry.

Feel free to have a look at the current database structure, we should be able to achieve this, I'm just not sure whether it is worth it.



P.S: Attempts for each stage/checkpoint during a run are already tracked and saved, we don't really track the amount of times you have beaten the map iirc or even how many tries it took you to beat it.
